components/sup/src/event/types.rs

```diff
@@ -105,6 +105,12 @@ impl EventMessage for ServiceStoppedEvent {
     }
 }
 
+impl EventMessage for ServiceUpdateStartedEvent {
+    fn event_metadata(&mut self, event_metadata: EventMetadata) {
+        self.event_metadata = Some(event_metadata);
+    }
+}
+
```

Comment:
If you're feeling ambitious and want to get into Rust macros, now might be a good time to tackle the suggestion on lines 93--94 😄

Reply:
thanks @cm, i'll check it out!

Reply:
done!